src/java.net.http/share/classes/jdk/internal/net/http/Http1Exchange.java

```diff
@@ -502,6 +502,16 @@ void cancel(IOException cause) {
         cancelImpl(cause);
     }
 
+    @Override
+    void onProtocolError(final String errorMessage) {
```

Comment:
Should take the IOException/ProtocolException as parameter and call cancelImpl(IOException). cancelImpl should do the right thing.